content/300-guides/050-database/800-production-troubleshooting.mdx

```diff
@@ -70,6 +70,136 @@ The following example demonstrates how to manually complete the steps of a migra
    prisma migrate resolve --applied "20201127134938_my_migration"
    ```
 
+## Fixing failed migrations with <inlinecode>migrate diff</inlinecode> and <inlinecode>db execute</inlinecode>
+
+In versions 3.9.0 and later, we are offering two new commands in [Preview](/about/prisma/releases#preview) that can be used to fix failed migrations.
+
+<Admonition type="warning">
+
+The `migrate diff` and `db execute` commands are currently a [Preview feature](/concepts/components/preview-features/cli-preview-features). To use these commands, you will need to add the `--preview-feature` flag.
+
+</Admonition>
+
+### New commands <inlinecode>migrate diff</inlinecode> and <inlinecode>db execute</inlinecode>
+
+The two new commands that we provide are:
+
+- `prisma migrate diff` which diffs two database schema sources to create a migration taking one to the state of the second. You can output either a summary of the difference or a sql script. The script can be output into a file via `> file_name.sql` or be piped directly to the new execute command.
+- `prisma db execute` which applies a SQL script to the database without interacting with the Prisma migrations table.
+
+### Example of a failed migration
+
+Imagine that you have the following `User` model in your schema, in both your local development environment and your production environment:
+
+```prisma file=schema.prisma
+model User {
+  id   Int    @id
+  name String
+}
+```
+
+At this point, your schemas in sync, but the data in the two environments is different.
+
+You then decide to make changes to your datamodel, adding another `Post` model and making the `name` field on `User` unique:
+
+```prisma file=schema.prisma
+model User {
+  id    Int     @id
+  name  String  @unique
+  email String?
+}
+
+model Post {
+  id    Int    @id
+  title String
+}
+```
+
+You create a migration called 'Unique' with `prisma migrate dev` which is saved in your local migrations history. Applying the migration succeeds in your dev environment and now it is time to release to production.
```

Comment:
Would be useful to have the command that creates a migration with the name 'Unique'

Reply:
That would then be `prisma migrate dev -n Unique` the one listed in the section would prompt for a name.

Reply:
Thanks, added this in